hs/src/LedgerState.hs

```diff
@@ -162,27 +172,49 @@ validInputs (TxWits tx _) l =
   where unspentInputs (UTxO utxo) = Map.keysSet utxo
 
 -- |Minimum fee calculation
-minfee :: PrtlConsts -> TxWits -> Coin
-minfee pc txwit = minfeeA pc * x + minfeeB pc
-  where x = Coin $ toEnum $ length (show txwit)
+minfee :: PrtlConsts -> Tx -> Coin
+minfee pc tx = minfeeA pc * x + minfeeB pc
```

Comment:
what is the reasoning for this change? Without looking at the witnesses in the minimum fee calculation, one can effectively create arbitrarily long transactions without additional fee. Or am I wrong here? We check that for each address in the set of input addresses, we have at least one signature. So here, one could have a valid transaction that adds one valid signature and lots of bogus signatures, too.

Reply:
after discussing this with others, we decided that we will have an abstract size function on transactions, which will be used for the minimum fee calculation. will we also enforce that (exactly) the minimal number of witness be used.

Reply:
ok, I've added a `txsize` function. In an earlier PR, @mgudemann , you mentioned that `Coin` might be the wrong units for the minfee calculation. Now I see better why it is weird, it doesn't make sense for x to be coins, it is a size. So I've changed this.

I made issue #109 for the two things we need to do with the current witnessing in the executable model.